packages/realm/src/errors.ts

```diff
@@ -103,11 +103,30 @@ export function fromBindingSyncError(error: binding.SyncError) {
   }
 }
 
+/**
+ * An class describing a sync error.
+ */
 export class SyncError extends Error {
   public name = "SyncError";
+
+  /**
+   * A numeric code representing the error.
+   */
   public code: number;
+
+  /**
+   * A string representing the error category.
+   */
   public category: string;
-  public isFatal: boolean;
+
+  /**
+   * The URL to the associated server log, if any.
```

Comment:
When you're saying "if any", do you mean that the URL could be empty? I interpret "if any" as if this field could be `undefined`.

Reply:
Most of the time `SyncError` can be caused by the server (for compensating writes, client resets and other errors), but also by the realm client itself (for example `illegal-realm-path` https://github.com/realm/realm-core/blob/11aa84592fc049b2a5492f80d725f7a202ca0a87/src/realm/sync/protocol.hpp#L302). In the last case there is no server logs because the error is raised locally.

Reply:
And yes, my bad, this can be empty but not undefined. I will update the docs to make it clearer.

Reply:
Alrighty thanks! 👍